### ISSUE TITLE
The latest_release plugin does not respect whitelist plugin

### ISSUE THREAD

**Issue by RWoodring79** [pypa/bandersnatch#239]
This could be a problem in my configuration, but I cannot get the latest_release and whitelist plugins to play nicely so I can include a specific older version of a package. I have the latest release filter set to 3 versions, and I tried to whitelist either 'urllib3' and 'urllib3-1.24.3' but neither worked. I know this topic was discussed as part of the latest_release filter issue or merge (cannot find it now), but the last I saw was it may have been pushed to later. Do you recall if it was fixed?

Just in case its a fault in my config, here is how I have my plugins configured:
```
[blacklist]
plugins =
    blacklist_project
    whitelist_project
    latest_release

packages =
    cntk
    . . .
    tf-estimator-nightly

[whitelist]
packages =
    urllib3-1.24.3

[latest_release]
keep = 3

```

**Comment by cooperlees** [comment]
Thanks for the report. I doubt this is your config, it's never been tested (the combination of all the plugins working together). I would expect whitelist, to negate your need for your blacklist tho. Why do you feel you need both?

PRs welcome here, I bet there is a logic bug somewhere.

**Comment by RWoodring79** [comment]
Perhaps I do not understand the purpose of the filters, but here is what I was thinking:
1) I started with the latest_release filter set to 3 versions to limit my storage needs. 
2) Then I blacklisted a bunch of cupy-cuda*, mxnet*, tensorflow*, tf-*, and *-nightly packages because they are either large or change frequently and we do not need them. 
3) Yesterday one of my developers needed urllib3-1.24.x which is older than latest_release includes, so I enabled the whitelist and added that one package.

**Comment by RWoodring79** [comment]
Can you expand on your "I would expect whitelist, to negate your need for your blacklist" comment? 

Also, does the order of the plugins in the config file matter? For example, would the latest_release remove the package I want before whitelist ever sees is, or would latest_release see/filter it regardless?

**Comment by cooperlees** [comment]
Whitelist is only at a package level and once enabled, you ONLY download those packages. Look out our CI job that runs bandersnatch, it uses a `whitelist`. Here is an example run:
- https://travis-ci.org/pypa/bandersnatch/jobs/534232059
Here is the config: https://github.com/pypa/bandersnatch/blob/master/src/bandersnatch/tests/ci.conf

Here we only ever download all of black, peerme and pyaib. It's also using the new [plugins] stanza for enabling plugins which will become default in 4.0 (no time frame on 4.0 yet).

Why can't your developers (who must use requests) pin it in their `requirements.txt` or equivalent. I feel bandersnatch plugins should be used to save disk and network bandwidth, not help with pinning for your specific versions. That will be a battle you'll never win.

**Comment by cooperlees** [comment]
@RWoodring79 - What's our next step here? Any more feedback. I don't know how I can action this.

**Comment by RWoodring79** [comment]
This was a misunderstanding on my part about how the whitelist plugin works. The feature I was imagining worked in conjunction with the recent releases plugin to allow me to hand pick specific older releases to include. For example, we are trying to run with only the 5 most recent releases but we have one project that needs urllib3 version that is 7 releases ago. 

When I first tried combining most recent and whitelist, I did not realize that whitelist was overriding the others (perhaps  big bold warning as such?). It was probably my fault for assuming there just were no packages that needed updated. 

I have not dug deep into the code, but the separated nature of each filter makes it seem to me a fairly significant overhaul of the plugin/filtering system might be required to make them play nicely together.
